### PR TITLE
[Export] Add an option to not save the weights

### PR DESCRIPTION
**Pull request by ManfeiBai**
add an option to not save the weights, and check whether the weights exist before load

### CODE REVIEW

File: torch_xla/stablehlo.py [hunk 1]
```diff
@@ -421,8 +434,10 @@ def _iter_dir(path: os.PathLike):
 
 def _load_program_bundle(stablehlo_dir: os.PathLike) -> StableHLOModelBundle:
   state_dict = {}
+  import os.path
```
Comment (qihqi) [inline torch_xla/stablehlo.py:437]: remove?

File: torch_xla/stablehlo.py [hunk 2]
```diff
@@ -421,8 +434,10 @@ def _iter_dir(path: os.PathLike):
...
   for name, f in _iter_dir(os.path.join(stablehlo_dir, 'data')):
-    state_dict[name] = np.load(f, allow_pickle=True)
+    if os.path.exists(f):
```
Comment (qihqi) [inline torch_xla/stablehlo.py:439]: put this outside of the for, 
check for `os.path.join(stablehlo_dir, 'data')`

File: torch_xla/stablehlo.py (hunk 1 above)
Comment (ManfeiBai) [reply torch_xla/stablehlo.py]: thanks, updated

File: torch_xla/stablehlo.py (hunk 2 above)
Comment (ManfeiBai) [reply torch_xla/stablehlo.py]: thanks, updated

File: test/stablehlo/test_stablehlo_save_load.py [hunk 3]
```diff
@@ -109,6 +122,18 @@ def test_save_load2(self):
...
+    model = ElementwiseAdd()
+    inputs = model.get_random_inputs()
+    options = StableHLOExportOptions()
+    options.override_tracing_arguments = inputs
+    options.save_weights = False
+    with tempfile.TemporaryDirectory() as tempdir:
+      save_torch_model_as_stablehlo(model, inputs, tempdir, options)
+      program2 = StableHLOGraphModule.load(tempdir)
+      result = program2(*inputs).detach().cpu()
+    self.assertTrue(torch.allclose(model(*inputs), result))
```
Comment (qihqi) [inline test/stablehlo/test_stablehlo_save_load.py:135]: Change these 2 lines to:
1. check that tempdir dont have weights
2. check that program2._bundle.state_dict is empty (no weights)

Comment (ManfeiBai) [reply test/stablehlo/test_stablehlo_save_load.py]: thanks, updated

File: torch_xla/stablehlo.py [hunk 4]
```diff
@@ -84,8 +84,8 @@ def get_stablehlo_text(self, method_name=None):
       method_name = self._default_method
     return self._name_to_stablehlo[method_name].text
 
-  def save(self, directory_path):
-    _save_program_bundle(self._bundle, directory_path)
+  def save(self, directory_path, options):
```
Comment (qihqi) [inline torch_xla/stablehlo.py:87]: options=None

**Review by qihqi — APPROVED** [review]
One more nit, overall good to go. Thanks!

File: torch_xla/stablehlo.py (hunk 4 above)
Comment (ManfeiBai) [reply torch_xla/stablehlo.py]: thanks, updated

**Pull request merged**